static/js/widget/GreetingWidget.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Created by newmember on 17/05/2017.
+ */
+
+define([
+	"dojo/_base/declare",
+	"dojo/_base/lang",
+	"dojo/dom-style",
+	'dojo/request/xhr',
+	"dojo/on",
+	"dojo/cookie",
+	"dijit/_WidgetBase",
+	"dijit/_TemplatedMixin",
+	"dojo/text!./templates/GreetingWidget.html"
+], function (declare, lang, domStyle, xhr, on, cookie, _WidgetBase, _TemplatedMixin, template) {
+	return declare("GreetingWidget", [_WidgetBase, _TemplatedMixin], {
+		author: "Anonymous",
+		avatarEdit: require.toUrl("../static/images/edit.png"),
+		avatarDelete: require.toUrl("../static/images/delete.png"),
+		templateString: template,
+		baseClass: "greetingWidget",
+
+		postCreate: function () {
+			this.inherited(arguments);
+			this._showAndHideEditContent(true);
+
+			this.own(
+					on(this.avatarEditNode, "click", lang.hitch(this, "_clickEdit")),
+					on(this.avatarDeleteNode, "click", lang.hitch(this, "_clickDelete")),
+					on(this.submitNode, "click", lang.hitch(this, "_clickSubmit"))
+			);
+		},
+
+		_clickEdit: function () {
+			this._showAndHideEditContent(false);
```

Comment:
nếu edit chỉ làm thao tác này, tại sao ko đưa hẳn vào trong `on(this.editNode, 'click',....)`